apiserver/facades/client/controller/controller.go

```diff
@@ -720,26 +721,57 @@ func (c *ControllerAPI) ModifyControllerAccess(args params.ModifyControllerAcces
 	hasPermission := err == nil
 
 	for i, arg := range args.Changes {
+		targetUserTag, err := names.ParseUserTag(arg.UserTag)
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(errors.Annotate(err, "could not modify controller access"))
+			continue
+		}
+
 		if !hasPermission {
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
+			// Security Event Logging: This log statement is required to comply with Canonical's SSDLC Security Event Logging policy.
+			// For details, see:
+			// https://library.canonical.com/corporate-policies/information-security-policies/ssdlc/ssdlc---security-event-logging
+			securitylog.LogAuthz(
+				securitylog.AuthzSecurityEvent{
+					Actor:  c.apiUser.Name(),
+					Target: targetUserTag.Name(),
```

Comment:
I think for all users, you're want the `Id` and not the `Name`. We'd want to know if it's an external user or not.

```suggestion
					Target: targetUserTag.Id(),
```

Reply:
yep, good catch about external user.